src/main/kotlin/calculator/domain/ExpressionFactory.kt

```diff
@@ -0,0 +1,29 @@
+package calculator.domain
+
+object ExpressionFactory {
+    private val DEFAULT_SEPARATORS = "[,:]".toRegex()
+    private val DEFAULT_EXPRESSION_TOKEN = listOf("0")
+    private val CUSTOM_DELIMITER_PATTERN = Regex("//(.)\n(.*)")
+
+    fun createExpression(input: String?): Expression =
+        parseExpression(
+            if (input.isNullOrEmpty()) {
+                DEFAULT_EXPRESSION_TOKEN
+            } else {
+                getTokens(input)
+            }
+        )
+
+    private fun getTokens(input: String): List<String> {
+        val result = CUSTOM_DELIMITER_PATTERN.find(input)
+        return result?.let {
+            val customDelimiter = it.groupValues[1]
+            require(customDelimiter.isNotEmpty()) { "커스텀 구분자가 비어있습니다." }
+            it.groupValues[2].split(customDelimiter)
+        } ?: input.split(DEFAULT_SEPARATORS)
+    }
```

Comment:
safe call(`?.`)과 elvis 연산자(`?:`)를 연속해서 사용하는 경우 자칫 잘못하면 `?. `와 `?:`에 있는 코드가 모두 실행될 가능성이 있습니다.
```kotlin
return something?.let {
    doSomething("1")
    null
} ?: doSomeThing("2")
```

이 경우는 elvis 연산만 잘 활용해도 코드 가독성을 확보할 수 있을 것 같아요!
```suggestion
    private fun getToken(input: String): List<String> {
        val result = CUSTOM_DELIMITER_PATTERN.find(input)
            ?: return input.split(DEFAULT_SEPARATORS)

        val customDelimiter = result.groupValues[1]
        
        require(customDelimiter.isNotEmpty()) { "커스텀 구분자가 비어있습니다." }
        
        return result.groupValues[2].split(customDelimiter)
    }
```

Reply:
크 👍 👍 👍  가독성이 확 좋아지네요 감사합니다!